h/util/group.py

```diff
@@ -4,7 +4,7 @@
 
 import re
 
-GROUPID_PATTERN = r'^group:([^@]+)@(.*)$'
+GROUPID_PATTERN = r"^group:([a-zA-Z0-9._\-+!~*()']{1,1024})@(.*)$"
```

Comment:
This is the only pattern used now for `groupid`. The group model does duplicate the `[a-zA-Z0-9._\-+!~*()']` part of this regex string for `authority_provided_id`, just so's yah know. It might be possible to consolidate this even more.

Reply:
One thing you'll note here is the length restriction to `{1,1024}`. This is because the model imposes maxlength of `1024` on `authority_provided_id`, but we have no way of knowing how long the rest of the `groupid` string is going to be (i.e. the `authority` part). So a maxlength failure on the `authority_provided_id` piece of a `groupid` will raise a 400, but the message will be about how it doesn't match the pattern, not length (the message does include the pattern for reference).